src/evaluate/hub.py

```diff
@@ -0,0 +1,63 @@
+from huggingface_hub.repocard import metadata_update
+
+
+def push_to_hub(
+    to: str,
+    task_type: str,
+    metric_value: float,
+    metric_type: str,
+    dataset_name: str,
+    dataset_type: str,
+    task_name: str = None,
+    dataset_config: str = None,
+    dataset_split: str = None,
+    dataset_revision: str = None,
+    dataset_args: dict[str, int] = None,
+    metric_name: str = None,
+    metric_config: str = None,
+    metric_args: dict[str, int] = None,
+):
+    """
+    TODO: Add documentation
+    """
+
+    result = {
+        "task": {
+            "type": task_type,
+        },
+        "dataset": {
+            "type": dataset_type,
+            "name": dataset_name,
+        },
+        "metrics": [
+            {
+                "type": metric_type,
+                "value": metric_value,
+            },
+        ],
+    }
+
+    if dataset_config:
```

Comment:
We should explicitly check if something is `None` with. E.g. `if dataset is not None:`.